src/UserSettingsPage.cpp

```diff
@@ -759,745 +761,858 @@ UserSettings::save()
-                    cameraResolutionCombo_->addItem(QString::fromStdString(resolution));
-            });
-
-    connect(cameraResolutionCombo_,
-            static_cast<void (QComboBox::*)(const QString &)>(&QComboBox::currentTextChanged),
-            this,
-            [this](const QString &resolution) {
-                settings_->setCameraResolution(resolution);
-                std::vector<std::string> frameRates = CallDevices::instance().frameRates(
-                  settings_->camera().toStdString(), resolution.toStdString());
-                cameraFrameRateCombo_->clear();
-                for (const auto &frameRate : frameRates)
-                    cameraFrameRateCombo_->addItem(QString::fromStdString(frameRate));
-            });
-
-    connect(cameraFrameRateCombo_,
-            static_cast<void (QComboBox::*)(const QString &)>(&QComboBox::currentTextChanged),
-            this,
-            [this](const QString &frameRate) { settings_->setCameraFrameRate(frameRate); });
-
-    connect(trayToggle_, &Toggle::toggled, this, [this](bool enabled) {
-        settings_->setTray(enabled);
-        if (enabled) {
-            startInTrayToggle_->setChecked(false);
-            startInTrayToggle_->setEnabled(true);
-            startInTrayToggle_->setState(false);
-            settings_->setStartInTray(false);
-        } else {
-            startInTrayToggle_->setChecked(false);
-            startInTrayToggle_->setState(false);
-            startInTrayToggle_->setDisabled(true);
-            settings_->setStartInTray(false);
-        }
-        emit trayOptionChanged(enabled);
-    });
-
-    connect(startInTrayToggle_, &Toggle::toggled, this, [this](bool enabled) {
-        settings_->setStartInTray(enabled);
-    });
-
-    connect(mobileMode_, &Toggle::toggled, this, [this](bool enabled) {
-        settings_->setMobileMode(enabled);
-    });
-
-    connect(groupViewToggle_, &Toggle::toggled, this, [this](bool enabled) {
-        settings_->setGroupView(enabled);
-    });
+QHash<int, QByteArray>
+UserSettingsModel::roleNames() const
+{
+    static QHash<int, QByteArray> roles{
+      {Name, "name"},
+      {Description, "description"},
+      {Value, "value"},
+      {Type, "type"},
+      {ValueLowerBound, "valueLowerBound"},
+      {ValueUpperBound, "valueUpperBound"},
+      {ValueStep, "valueStep"},
+      {Values, "values"},
+      {Good, "good"},
+      {Enabled, "enabled"},
+    };
 
-    connect(decryptSidebar_, &Toggle::toggled, this, [this](bool enabled) {
-        settings_->setDecryptSidebar(enabled);
-    });
+    return roles;
+}
 
-    connect(privacyScreen_, &Toggle::toggled, this, [this](bool enabled) {
-        settings_->setPrivacyScreen(enabled);
-        if (enabled) {
-            privacyScreenTimeout_->setEnabled(true);
-        } else {
-            privacyScreenTimeout_->setDisabled(true);
+QVariant
+UserSettingsModel::data(const QModelIndex &index, int role) const
```

Comment:
I largely hate this, but don't have suggestions short of a rewrite.

Reply:
Rude!

Reply:
And I actually think it is slightly cleaner, since at least it now mostly separates representation and function. There are some ways it can be cleaned up, but that can be done later.